### PR TITLE
Ap 14 upstream downstream sops repo

### PR DESCRIPTION
**Pull request by R-Lawton**

### CODE REVIEW

**Comment by R-Lawton** [comment]
@david-martin @pepedocs @emmanuelbernard @brizrobbo @jewzaam @tombentley This is the AP created based on [ADR 85](https://github.com/bf2fc6cc711aee1a0c2a/architecture/blob/main/_adr/85/index.adoc) if ye get time to have a look that would be great

File: _ap/14/index.adoc
Comment (emmanuelbernard) [inline _ap/14/index.adoc:10]: Minor but if you could break line at the end of each sentence, it makes for a more diff readable / editable content.

**Review by emmanuelbernard — CHANGES_REQUESTED** [review]

File: _ap/14/index.adoc
Comment (brizrobbo) [reply _ap/14/index.adoc:4]: Agreed. Of course 'sensitive' data should always be prevented from leaking :), but this is more about "enabling the consumers of the SOPs to have more context-specific SOPs for their own purposes. E.g. SREs at a particular company may need a specific task that only makes sense for that company.

File: _ap/14/index.adoc
Comment (brizrobbo) [reply _ap/14/index.adoc:10]: I would change "mitigate exposing sensitive content" to focus more on "enabling suitable content per context". See my previous comment for a more detailed explanation for this opinion.

File: _ap/14/index.adoc
Comment (brizrobbo) [reply _ap/14/index.adoc:14]: Again, remove "accidental leaks" for a more positive veiwpoint.

**Review by brizrobbo — CHANGES_REQUESTED** [review]

File: _ap/14/index.adoc
Comment (pepedocs) [reply _ap/14/index.adoc:4]: @emmanuelbernard I believe we are trying to mitigate the potential security threat here that was raised in the ADR document. 

I believe as we have open sourced this SOP process, we also inadvertently opened potential issues such as this security threat of any committer committing private data and therefore we must be very clear about this threat and provide an example on how to mitigate them.

Comment (R-Lawton) [reply _ap/14/index.adoc:4]: Ok yeah that makes sense ill make that change

Comment (R-Lawton) [reply _ap/14/index.adoc:4]: Im not sure i follow the discussion here and maybe i misunderstood the original ask for the AP, but are we not trying to document a easily repeatable model that gives a guide to other Red Hat services on how they could create a upstream repo with steps on how to lower the possibility of exposing Red Hat internal information?

Comment (pepedocs) [reply _ap/14/index.adoc:4]: >document a easily repeatable model that gives a guide to other Red Hat services on how they could create a upstream repo with steps on how to lower the possibility of exposing Red Hat internal information

@R-Lawton Just to clarify that this is my understanding and I would recommend doing this as it seems it is our responsibility to provide at least an example mitigation model.

Comment (pepedocs) [reply _ap/14/index.adoc:4]: > So would it be ok for everyone to widen the scope of the AP as I mentioned in the title and explicitly list use cases somewhere and including the sensitive data aspect?

It will list the example mitigation case so it's ok. I agree.

Comment (tombentley) [reply _ap/14/index.adoc:4]: The _value_ here isn't so much in "preventing leaks" of internal details (though that _is_ a benefit). Instead it could be very valuable for upstream projects (such as, but certainly not limited to, Strimzi) to maintain SOPs for their software, simply because it's of direct benefit to their users. This allows things like:

* Having these SOPs tested and maintained upstream, where it's least costly
* Letting people in those upstream communities benefit from, contribute to & improve those SOPs, and develop new ones
* Keeping the knowledge of how to effectively operate the software in the community which produces the software.

Where those upstream communities are also upstreams for other Red Hat products then some of this value also flows back to other Red Hat stuff, but I think the case can be made for doing this upstream because it's simply the right thing to do.

**Review by tombentley — COMMENTED** [review]
Rather than spending a lot of time and effort in describing an ideal of what we'd like, I wonder whether a useful step at this point would be to try to move the existing Strimzi-related SOPs to the Strimzi documentation, and watching how that works its way through the Strimzi review process. If it's successful it makes it easier to write this document because we can point to merged PRs, and explain it all more concretely.

**Comment by tombentley** [comment]
> Firstly nearly all if not all of our Strimzi SOPs are very RHOSAK specific they would essentially have to be completely re-written to fit with purely Strimzi docs.

What makes them so RHOSAK-specific? I would have thought that the actions needed to address Kafka-specific things like under-replicated partitions, would be pretty easy RHOSAK-independent.

I completely accept it might not be trivial to move them, and it might need to be done on a SOP-by-SOP basis. And it would probably involve working with the upstream community (in this case I guess @PaulRMellor might be a good point of contact). 

Understanding the answers to these questions is kind of central to understanding how easily Strimzi can make use of the operational learnings that RHOSAK is making. It's quite possible that the answer is "not easily", but understanding _why_ is an important part of the process. 

> Would that mean then RHOSAK upstream SOP repo wouldn't have any Strimzi SOPs in it?

Eventually maybe. Would that be a problem?

**Comment by emmanuelbernard** [comment]
@tombentley my perspective is that you aim is good but you might aim for too far for this evolution step and too abstract to keep the ball rolling.
My proposal is to go through with this rev and have subsequently Strimzi + RHOSAK explore how some SOPs could move upstream and find the obstacles. With that feedback we can next rev that architecture pattern. 
In short, lets go with concrete examples before capturing.

**Comment by R-Lawton** [comment]
Ok just so im clear, the plan is to continue with this AP with the topic being ADR 85 but in a AP form i.e how a team can implement a model for maintaining an upstream version of SOPs, and a mechanism to consume them downstream from the point of view that its not just Red Hat

**Comment by emmanuelbernard** [comment]
@R-Lawton What is preventing you at the moment to reach the closing state on this one?

**Comment by R-Lawton** [comment]
Apologies, im just finishing off my latest version off it. Ill have it finished later today

File: _ap/14/index.adoc
Comment (tombentley) [inline _ap/14/index.adoc:17]:
```suggestion
The recommended flow of using information contributed upstream repo in the internal downstream repo is through the use of automation. 
```

File: _ap/14/index.adoc
Comment (tombentley) [inline _ap/14/index.adoc:22]:
```suggestion
Some pitfalls that contributors have to be mindful of are,
```

**Review by tombentley — APPROVED** [review]
Approved, since I don't want to be a blocker here. However, I think it's possible that Strimzi (as an example of an upstream project) might in the future end up with its own SOP-like documents which would be duplicating some of this content. That's seems inevitable at this point, give the requirements here.

**Comment by R-Lawton** [comment]
@emmanuelbernard @brizrobbo are ye also happy for this to be merged?

**Review by emmanuelbernard — APPROVED** [review]

**Comment by emmanuelbernard** [comment]
+! @R-Lawton push any time you want. I can push if you want me to

**Comment by R-Lawton** [comment]
i dont have write access to merge so you will have to do the honours

**Comment by R-Lawton** [comment]
i would like @brizrobbo to approve to before we merge

**Review by pepedocs — APPROVED** [review]

**Comment by brizrobbo** [comment]
LGTM

**Review by brizrobbo — APPROVED** [review]

**Pull request merged**
